File: 2016/day1/pa.lua
local x, y = 0, 0
local facing = 0 -- n0 e1 s2 w3

local result = {}

for line in io.lines("./input.txt") do
	for value in string.gmatch(line, "[^,]+") do
		value = value:match("^%s*(.-)%s*$")
		table.insert(result, value)
	end
end

for _, v in ipairs(result) do
	local turn, steps = v:match("^(%a)(%d+)$")
	steps = tonumber(steps)

	if turn == "R" then
		facing = (facing + 1) % 4
	elseif turn == "L" then
		facing = (facing + 3) % 4
	else
		error("Invalid turn: " .. turn)
	end

	if facing == 0 then
		y = y + steps
	elseif facing == 1 then
		x = x + steps
	elseif facing == 2 then
		y = y - steps
	elseif facing == 3 then
		x = x - steps
	end
end

local distance = math.abs(x) + math.abs(y)
print(distance)

File: 2016-py/day1/pa.py
def main():
    x, y = 0, 0
    facing = 0  # north=0, east=1, south=2, west=3

    result = []

    with open("input.txt", "r") as fin:
        for line in fin:
            values = line.strip().split(",")
            for value in values:
                result.append(value.strip())

    for instruction in result:
        turn = instruction[0]
        steps = int(instruction[1:])

        if turn == "R":
            facing = (facing + 1) % 4
        elif turn == "L":
            facing = (facing + 3) % 4
        else:
            raise ValueError(f"Invalid turn: {turn}")

        if facing == 0:
            y += steps
        elif facing == 1:
            x += steps
        elif facing == 2:
            y -= steps
        elif facing == 3:
            x -= steps

    distance = abs(x) + abs(y)
    print(distance)


if __name__ == "__main__":
    main()

